server/src/main/java/io/crate/exceptions/SQLExceptions.java

```diff
@@ -91,6 +92,22 @@ public static Throwable unwrap(@Nonnull Throwable t) {
         return unwrap(t, null);
     }
 
+    public static String detailedMessageOf(@Nullable Throwable t) {
```

Comment:
This would provide a message in the following format e.g.  `RelationUnknown 'Relation 'sys.nodes' unknown' thrown in io.crate.analyze.relations.FullQualifiedNameFieldProvider.resolveField(FullQualifiedNameFieldProvider.java:113)'`. The only downside here is that the position could change based on the code, so for unit testing, we could either go with startsWith, or provide a message with a the additional information. The tests need to be adapted, which i will do once we finalise the error message.

Reply:
That's not what I meant. 

My intention was to keep the original logic:

```java
String message = throwable.getMessage();
if (message == null) {
    if (throwable instanceof CrateException && throwable.getCause() != null) {
        throwable = throwable.getCause();   // use cause because it contains a more meaningful error in most cases
    }
    StackTraceElement[] stackTraceElements = throwable.getStackTrace();
    if (stackTraceElements.length > 0) {
        message = String.format(Locale.ENGLISH,
                                "%s in %s",
                                throwable.getClass().getSimpleName(),
                                stackTraceElements[0]);
    } else {
        message = "Error in " + throwable.getClass().getSimpleName();
    }
}
```


and use that in both, the HTTP and PostgreSQL case.


Note that:

- This only populates the `message`
- It only uses the stacktrace information if the `message` of the throwable is null.



Basically, I think if we extended the `if (t == null)` case in `messageOf` to utilize the stacktrace, we could just use `messageOf(t)` in both HTTP and PostgreSQL and not need any new method.

Reply:
Ah, thanks for explaining 👍

Reply:
I hope this version makes sense now.